spit/classifier.py

```diff
@@ -141,6 +142,152 @@ def compare_with_best(self, test_dset, test_labels, file_path):
 
     return
 
+  def _train(self, epochs, batch_size, subset_percent=None, train_images=None, train_labels=None, validation_data=None, steps_per_epoch=None, validation_freq=1, test_model=None, spit_path=os.getenv('SPIT_PATH'), save_path=os.getenv('SAVE_PATH')):
+    """
+
+    Trains the classifier with given images, labels, and training parameters.
+
+    Parameters:
+
+    :param epochs:
+      Number of epochs of the training. 
+      Must be an integer value.
+
+    :param batch_size:
+      Size of the training batches formed in the process. 
+      Must be an integer value.
+
+    :param save_path:
+      Path to where the best model will be saved.
+
+    :param train_images:
+       Set of images for model to train on.
+       Assume this is a numpy array with (batch_size, width, height, num_channels) as its dimensions.
+
+    :param train_labels:
+      Set of test labels corresponding to test images.
+      Assume this is a rank 1 array with (batch_size, ) as its dimensions.
+
+    :param validation_data:
+      Data to be used for the validation set. 
+      Assume this is a tuple with (images, labels) with same dimensions as train_images, train_labels.
+      If None is specified, validation_data will be None.
+
+    :param steps_per_epoch:
+      Total number of steps (batches of samples) before declaring one epoch finished and starting the next epoch.
+      Assume this is an integer value. If None is specified, this will be None.  
+
+    :param validation_freq:
+      Specifies how many training epochs to run before a new validation run is performed.
+      Assume this is an integer value or a collection containing the epochs at which to run validation (ie [1,2,10]).
+
+    :param test_model:
+      An alternative choice for model to train on. Assume None.
+
+    :param spit_path & param save_path:
+      Path to the spit images and where the model will be saved respectively.
+      ***Environmental variables must be set by caller or the path must be passed manually.***
+
+    Returns:
+    :returns history:
+      Tensorflow History object containing loss and accuracy data over the training.
+
+    """
+    # choose the model
```

Comment:
While at first it made some sense to me to have this be a method under `Classifier`, these lines make it obvious that we want to use this code on any number of models.  Therefore, I think we should make this a stand-alone method in the `train.py` module.

Reply:
Further, it isn't as if we access the model internals (i.e. `self`) in any sigfnificant way in the remainder of this method

Reply:
We can and should make `split_array` a method of `train.py` as well.

Reply:
Agreed, moved and will submit new request.